tensorflow/core/graph/graph.cc

```diff
@@ -453,6 +453,20 @@ const Edge* Graph::AddControlEdge(Node* source, Node* dest,
   return AddEdge(source, kControlSlot, dest, kControlSlot);
 }
 
+void Graph::RemoveControlEdge(const Edge* e) {
+  if (!e->src_->IsSource() && !e->dst_->IsSink()) {
+    e->dst_->MaybeCopyOnWrite();
+    auto* inputs = e->dst_->props_->node_def.mutable_input();
+    for (auto it = inputs->begin(); it != inputs->end(); ++it) {
+      if (*it == strings::StrCat("^", e->src_->name())) {
```

Comment:
nit: pull the strings::StrCat out of the for loop so you're not computing it every time (you could even not construct a string at all, but it's fine)

Reply:
Good catch. :)